active/CornerRadius/CornerRadiusFilterConverter.md

```diff
@@ -0,0 +1,37 @@
+# Background
+
+Some controls need to turn on/off part of their rounded corners depending on their states.
+
+e.g. When CommandBarFlyout opens up its secondary menu, primary and secondary menus are clipped together we need to clear rounded corners on bottom of primary menu and top of secondary menu to close the gaps.
+
+![CommandBarFlyout CornerRadius example](images/commandbarflyout-cornerradius-example.png)
+
+You can't set individual values of a CornerRadius object in markup according to our [doc](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.CornerRadius#notes-on-xaml-syntax). We need a way to get non-uniform corner radius values to implement the design above.

```

Comment:
You can set the components of a CornerRadius, for example:
```
<Border ...>
    <Border.CornerRadius>
        <CornerRadius>4,4,0,0</CornerRadius>
    </Border.CornerRadius>
</Border>

Reply:
Yes, but what we need is a way to extract values from uniform CornerRadius. The problem we were facing was, if user passes in CornerRadius="x", how do we get non-uniform values like x,x,0,0 just using markup? 

Take FlipView as example, given `<FlipView CornerRadius="4"/>`, we need to set 0,4,4,0 for horizontal previous button and 4,0,0,4 for horizontal next button.
![{81A9E3CE-45F5-43CC-903E-937AB93A4090}](https://user-images.githubusercontent.com/4424330/62188949-a8935a80-b322-11e9-8e46-a37b2010bbb2.png)

If setting individual values is possible we can probably do 
```XAML
<CornerRadius x:Key="HorizontalPreviousCornerRadius"
    TopLeft="0"
    TopRight="{Binding RelativeSource={RelativeSource TemplatedParent}, 
    Path=CornerRadius.TopLeft}"
    BottomRight="{Binding RelativeSource={RelativeSource TemplatedParent}, 
    Path=CornerRadius.BottomRight}"
    TopRight="0"/>
```
But looks like that's not supported so we came up with this converter idea.

Reply:
I see, so the motivating use case is in control templates. Does that prevent usage of a method binding?
```
CornerRadius="{x:Bind FilterCorner(CornerRadius, FilterCornerLocation.Top)}"
```

Reply:
The problem with that is the generated code has no place to live since generic.xaml has no code-behind. I tried x:bind first and it didn't compile :\

We had 3 options just like you mentioned in the review, x:bind to a function, TemplateSettings and converter. Converter was the last thing we wanted to do but with all the restrictions seems like that's the only option that works.